api/collections/views.py

```diff
@@ -0,0 +1,283 @@
+from rest_framework import generics, permissions as drf_permissions
+from modularodm import Q
+from datetime import datetime
+
+from framework.auth.core import Auth
+from website.models import Node, Pointer
+from api.base.utils import get_object_or_404
+from api.base.filters import ODMFilterMixin, ListFilterMixin
+from .serializers import CollectionSerializer, CollectionPointersSerializer
+from .permissions import ContributorOrPublic
+from website.views import find_dashboard
+
+smart_folders = {
+    '~amp': Q('category', 'eq', 'project') & Q('is_deleted', 'eq', False) &
+    Q('is_registration', 'eq', False) & Q('is_folder', 'eq', True) &
+    Q('__backrefs.parent.node.nodes', 'eq', None),
+
+    '~amr': Q('category', 'eq', 'project') & Q('is_deleted', 'eq', False) &
+    Q('is_registration', 'eq', True) & Q('is_folder', 'eq', True) &
+    Q('__backrefs.parent.node.nodes', 'eq', None),
+}
+
+
+class CollectionMixin(object):
+    """Mixin with convenience methods for retrieving the current node based on the
+    current URL. By default, fetches the current node based on the pk kwarg.
+    """
+
+    serializer_class = CollectionSerializer
+    node_lookup_url_kwarg = 'pk'
+
+    def get_node(self):
+        key = self.kwargs[self.node_lookup_url_kwarg]
+        if key == 'dashboard':
+            return find_dashboard(self.request.user)
+
+        if key == 'amr' or key == 'amp':
+            ret = {
+                'id': '{}'.format(key),
+                'pk': key,
+                'title': "Smart Folder {}".format(key),
```

Comment:
A smart folder is still a folder, so include the is_folder property here as well. I presume if is_dashboard exists, it will return falsy in the serializer, so if it doesn't have a problem, that's probably fine.